app/src/main/java/com/brandon/forzenbook/view/composables/LoginComposables.kt

```diff
@@ -53,85 +48,110 @@ fun LoginContent(
         }
-        BlackButton(text = resources.getString(R.string.loginButtonText)) {
+        BlackButton(
+            text = if (state.isCodeSent) resources.getString(R.string.loginButtonText) else resources.getString(
+                R.string.loginGetCodeText
+            )
+        ) {
             keyboardController?.hide()
-            if (username.isNotEmpty() && code.isNotEmpty()) {
-                onSubmit(username, code)
-            } else {
-                emptyFieldError = true
-            }
-        }
-        if (emptyFieldError) {
-            TextFieldErrorText(text = resources.getString(R.string.requiredFieldsError))
+            if (onCheckConnection()) {
+                if (state.isCodeSent) {
+                    if (email.isNotEmpty() && code.isNotEmpty()) {
+                        emptyFieldError = false
+                        onLogin(email, code)
+                    } else emptyFieldError = true
+                } else {
+                    if (email.isNotEmpty()) {
+                        emptyFieldError = false
+                        onGetCode(email)
+                    } else emptyFieldError = true
+                }
+            } else isDialogOpen = true
         }
+        if (emptyFieldError) TextFieldErrorText(text = resources.getString(R.string.requiredFieldsError))
         RedirectText(text = resources.getString(R.string.loginCreateAccount)) {
             navController?.navigate(NavigationDestinations.CREATE_ACCOUNT) {
                 popUpTo(NavigationDestinations.CREATE_ACCOUNT) { inclusive = true }
             }
         }
     }
+    if (isDialogOpen) {
+        AlertDialog(
+            title = resources.getString(R.string.errorNoInternetConnection),
+            body = resources.getString(R.string.errorConnectAndTryAgain),
+            onDismissRequest = { isDialogOpen = false },
+        )
+    }
 }
 
 @Composable
-fun LoginTitle() {
-    Image(
-        painter = painterResource(id = R.drawable.forzenlogo),
-        contentDescription = LocalContext.current.getString(R.string.loginImageDescription),
-        modifier = Modifier.size(250.dp)
-    )
-    TitleText(
-        text = LocalContext.current.getString(R.string.loginTitle)
-    )
+fun LoadingLoginScreen(
+    state: Loading
+) {
+    val resources = LocalContext.current.resources
+    var email by rememberSaveable { mutableStateOf(state.email) }
+    var code by rememberSaveable { mutableStateOf(TEXT_FIELD_DEFAULT_TEXT) }
+    YellowColumn {
+        ImageTitle(
+            imageId = R.drawable.forzenlogo,
+            text = resources.getString(R.string.loginTitle),
+        )
+        InputInfoTextField(
+            hint = resources.getString(R.string.loginEmailHint),
+            currentText = email,
+            isEnabled = false,
+            onTextChange = { email = it },
+            characterLimit = EMAIL_CHAR_LIMIT,
+            onMaxCharacterLength = {},
```

Comment:
What's the default for this? Why include?

Reply:
removed.